### PR TITLE
FOGL-939, FOGL-940: Add JQ filter configuration on north plugins and update sending process

### PR DESCRIPTION
**Pull request by Singhal-Vaibhav**
~~Looked from http receiving side:~~
~~Earlier response (where we are manipulating each read tuple in code):~~
```
[{'asset_code': 'fogbench/pressure', 'readings': [{'reading': {'pressure': 901.3}, 'user_ts': '2018-01-11 14:13:18.157493+05:30', 'read_key': 'c452a4e1-c9ed-4023-92ee-ca37d0c354dc'}]}, {'asset_code': 'fogbench/humidity', 'readings': [{'reading': {'temperature': 20, 'humidity': 14}, 'user_ts': '2018-01-11 14:13:18.157535+05:30', 'read_key': 'd0f364e4-f43b-45cd-8553-01d0715c8a27'}]}]
```

~~Using JQ Filter:~~
```
[{'asset_code': 'fogbench/pressure', 'readings': [{'reading': {'pressure': 901.3}, 'user_ts': '2018-01-11 14:13:18.157493+05:30', 'read_key': 'c452a4e1-c9ed-4023-92ee-ca37d0c354dc'}]}, {'asset_code': 'fogbench/humidity', 'readings': [{'reading': {'temperature': 20, 'humidity': 14}, 'user_ts': '2018-01-11 14:13:18.157535+05:30', 'read_key': 'd0f364e4-f43b-45cd-8553-01d0715c8a27'}]}]
```

~~No change in output~~

As per the latest discussion, the scope of using jq filter is manipulating in data only. Plugins assume that they get the right set of keys and payloads keys are not altered. Default jq configuration will be False

### CODE REVIEW

File: python/foglamp/plugins/north/http_north/http_north.py [hunk 1]
```diff
@@ -43,6 +44,16 @@
         'description': 'how long (x seconds) the plugin should wait for pending tasks to complete or cancel otherwise',
         'type': 'integer',
         'default': '10'
+    },
+    "applyFilter": {
+        "description": "Whether to apply filter before processing the data",
+        "type": "boolean",
+        "default": "True"
```
Comment (MarkRiddoch) [inline python/foglamp/plugins/north/http_north/http_north.py]: Before we check in I think we should make the filter disabled by default

**Comment by praveen-garg** [comment]
a quick question: if we only change filter rule from config; shall not mapping for send block shall break?

**Comment by Singhal-Vaibhav** [comment]
@praveen-garg Yes and No :) , 
If we set applyFilter to true and change the filterRule for any plugin, it depends on what the filter rule is. If the filter rule manipulates the keys of the payload dictionary the send block of plugins will break as currently both the omf and http_north are manipulating the reading block based on the payload keys. 
On the other hand, we use filter to manipulate the data of the payload (without touching the keys), then it will not break. e.g's can be adding a unit to the data, and so on.

File: python/foglamp/plugins/north/http_north/http_north.py (hunk 1 above)
Comment (izoratti) [reply python/foglamp/plugins/north/http_north/http_north.py]: I agree, the filter must be disabled.

**Pull request merged**